pagopa/io-app
# chore(Cross): [IOAPPX-435] Improve overall UI typography + Update `io-app-design-system` to `4.0.1`

## Short description
This PR improves overall UI typography.

## List of changes proposed in this pull request
- Update `io-app-design-system` to the new major `4.0.1` version
- Update font weight in the `HeaderFirstLevel` to make the title bolder
- Improve **Strong** markdown style by replacing `Bold` with the `Semibold` style
- Decrease font weight of the subtitle in the `ServicesHeader` component
- Improve formatting of the `MessageListItem`
- Replace removed `Chip` typographic style with `LabelMini`
- Replace removed `Label` typographic style with `Body`
- Replace removed `LabelSmall` typographic style with the new `BodySmall`
- **[extra]** Fix typo in the English locale (`CIEIdWizard`)

> [!note]
> **For the reviewers:** the default weight of `LabelSmall` was `Semibold`, while the default weight for `BodySmall` is `Regular`. That's why `LabelSmall` set without the `weight` prop was replaced by `BodySmall` with the `weight` prop  set to `Semibold`. The same should happen to the `Label` typographic style

### Preview
| Actual UI | New UI |
|--------|--------|
| ![Simulator Screenshot - iPhone 16 Pro - 2024-11-21 at 11 47 43](https://github.com/user-attachments/assets/70ddf413-a5a7-40be-ac54-d8d1d81ce040) | ![Simulator Screenshot - iPhone 16 Pro - 2024-11-21 at 11 52 44](https://github.com/user-attachments/assets/f1616c82-bd68-49f7-bda0-92947039b79f) | 

#### Old vs New

<details><summary><code>Messages Home</code></summary>
<p>

| Before | After |
|--------|--------|
| ![Simulator Screenshot - iPhone 16 Pro - 2024-12-05 at 15 19 58](https://github.com/user-attachments/assets/13dcd29c-a097-4fde-a010-c1517a73a299) | ![Simulator Screenshot - iPhone 16 Pro - 2024-12-05 at 15 22 33](https://github.com/user-attachments/assets/f10f92c9-8b49-4827-adac-9ed0d73561ac) | 

</p>
</details> 

<details><summary><code>Payments Home</code></summary>
<p>

| Before | After |
|--------|--------|
| ![Simulator Screenshot - iPhone 16 Pro - 2024-12-05 at 15 20 22](https://github.com/user-attachments/assets/10cae2b8-c71b-42ac-8e27-f3f328ab4bd8) | ![Simulator Screenshot - iPhone 16 Pro - 2024-12-05 at 15 23 51](https://github.com/user-attachments/assets/0ad36333-7cdc-4d0a-af13-cc442bb46577) | 

</p>
</details> 

### Related PRs
* https://github.com/pagopa/io-app-design-system/pull/360
* https://github.com/pagopa/io-app-design-system/pull/361

## How to test
Check each DS component and navigate through the app to check if everything is OK
